File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        vector<int> left, middle, right;

        for (int num : nums) {
            if (num < pivot) {
                left.push_back(num);
            } else if (num == pivot) {
                middle.push_back(num);
            } else {
                right.push_back(num);
            }
        }

        left.insert(left.end(), middle.begin(), middle.end());
        left.insert(left.end(), right.begin(), right.end());

        return left;
    }
};

File: 2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cs
public class Solution {
    public int[] PivotArray(int[] nums, int pivot) {
        int[] result = new int[nums.Length];
        int left = 0, right = nums.Length - 1;
        
        for (int i = 0, j = nums.Length - 1; i < nums.Length; i++, j--) {
            if (nums[i] < pivot) {
                result[left] = nums[i];
                left++;
            }
            if (nums[j] > pivot) {
                result[right] = nums[j];
                right--;
            }
        }
        
        while (left <= right) {
            result[left] = pivot;
            left++;
        }
        
        return result;
    }
}